addon/components/sl-split-grid.js

```diff
@@ -0,0 +1,564 @@
+            }
+        }
+    },
+
+    // -------------------------------------------------------------------------
+    // Events
+
+    // -------------------------------------------------------------------------
+    // Properties
+
+    /**
+     * The text label for the rows' actions buttons
+     *
+     * @property {string} actionsButtonLabel
+     * @default  "Actions"
+     */
+    actionsButtonLabel: 'Actions',
+
+    /**
+     * The width of the actions column, in pixels
+     *
+     * @property {number} actionsColumnWidth
+     * @default  120
+     */
+    actionsColumnWidth: 120,
+
+    /**
+     * The row record that is currently active in the detail pane
+     *
+     * @property {object} activeRecord
+     * @default  null
+     */
+    activeRecord: null,
+
+    /**
+     * Determines whether the height of the split-grid is reactive to viewport
+     *
+     * When true, the component's content areas will be automatically resized
+     * to the available viewport height when the viewport changes.
+     *
+     * @property {boolean} autoHeight
+     * @default  true
+     */
+    autoHeight: true,
+
+    /**
+     * The height of the split-grid's detail pane content, in pixels
+     *
+     * @property {number} detailContentHeight
+     * @default  600
+     */
+    detailContentHeight: 600,
+
+    /**
+     * The name of the controller/template/view to use for the detail pane's footer section
+     *
+     * @property {string} detailFooterTemplate
+     * @default  null
+     */
+    detailFooterTemplate: null,
+
+    /**
+     * The name of the controller/template/view to use for the detail pane's header section
+     *
+     * If this value is null (default), then the detail pane's header will be
+     * populated by text determined by the `detailTitlePath` attribute.
+     *
+     * @property {string} detailHeaderTemplate
+     * @default  null
+     */
+    detailHeaderTemplate: null,
+
+    /**
+     * The name of the controller/template/view to use for the detail pane content
+     *
+     * The controller matching this name also drives the data for the detail's
+     * footer template and header template.
+     *
+     * @property {string} detailContent
+     * @default  null
+     */
+    detailName: null,
+
+    /**
+     * Indicates when the details pane is open
+     *
+     * @property {boolean} detailsOpen
+     * @default  false
+     */
+    detailsOpen: false,
+
+    /**
+     * The name of the controller/template/view to use for the filter panel
+     *
+     * @property {string} filterName
+     * @default  null
+     */
+    filterName: null,
+
+    /**
+     * Indicates when the filter pane is open
+     *
+     * @property {boolean} filterOpen
+     * @default  false
+     */
+    filterOpen: false,
+
+    /**
+     * The text to display on the filter panel toggle button
+     *
+     * @property {string} filterText
+     * @default  "Filter"
+     */
+    filterText: 'Filter',
```

Comment:
This needs to be abstracted or definable via component usage to support translation needs